### DISCUSSION TITLE
Benchmarking and user facing repository

### DISCUSSION THREAD

**Discussion by reubenharry** [blackjax-devs/blackjax#702] [General]
My understanding is that blackjax is a developer facing library, and while it is intended to be easy to use, the focus is not on maximally simple user facing code. For example, we don't have a simple wrapper of the sampler and tuning for a given algorithm, like `blackjax.mcmc.nuts(logdensity=..., num_steps=..., tuning='standard', num_chains=...)` and instead have to write a function that does this (as in the example here https://blackjax-devs.github.io/sampling-book/algorithms/mclmc.html).

However, I want to find a way to give users very easy access to the samplers (e.g. MCLMC, which I'm involved in developing). I would also like to offer a simple way of producing comparisons of samplers' ESS across a range of problems. I'm wondering if the best option is to make a new repo that does these two things (in practice, this is what I am currently doing, but haven't polished and released that code). Or do people think that there's an existing solution in the jax space already?

**Comment by junpenglao** [comment]
🤖 *This reply was drafted with Claude and reviewed by me (@junpenglao).*

Soft-launch: we now have **tuningfork** — a BlackJAX-native benchmark library — public at https://github.com/blackjax-devs/tuningfork. It's still WIP but coming together (a model suite × base methods × warmup/SMC strategies). It's the home for exactly this kind of standardized benchmarking plus higher-level "just run the sampler" ergonomics. Feedback welcome.
